components/data_comps/datm/bld/build-namelist

```diff
@@ -381,9 +381,7 @@ if ( $GRID eq "CLM_USRDAT" && $CLM_USRDAT_NAME =~ /^(UNSET|)$/ ) {
 EOF
 }
 
-(-d $DIN_LOC_ROOT)  or  die <<"EOF";
-** $ProgName - CESM inputdata root is not a directory: \"$DIN_LOC_ROOT\" **
-EOF
+(-d $DIN_LOC_ROOT)  or mkdir $DIN_LOC_ROOT;
```

Comment:
Yes, that's it.

On Thu, Jan 7, 2016 at 12:57 PM, Doug Jacobsen <notifications@github.com>
wrote:

> In components/data_comps/datm/bld/build-namelist
> <https://github.com/ESMCI/cime/pull/18#discussion_r49118186>:
>
> > @@ -381,9 +381,7 @@ if ( $GRID eq "CLM_USRDAT" && $CLM_USRDAT_NAME =~ /^(UNSET|)$/ ) {
> >  EOF
> >  }
> >
> > -(-d $DIN_LOC_ROOT)  or  die <<"EOF";
> > -** $ProgName - CESM inputdata root is not a directory: \"$DIN_LOC_ROOT\" **
> > -EOF
> > +(-d $DIN_LOC_ROOT)  or mkdir $DIN_LOC_ROOT;
>
> So, for example, it should be done in the setup script, or in the build
> script, rather than in build-namelist, correct?
>
> I'd be fine with either of those, in which case you're right this should
> error, since build-namelist called by things that should have created the
> directory.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ESMCI/cime/pull/18/files#r49118186>.
>



-- 
Jim Edwards

CESM Software Engineer
National Center for Atmospheric Research
Boulder, CO